### PR TITLE
4 x

### PR DESCRIPTION
**Pull request by agentrickard**
## Ticket(s)

- See ticket https://www.drupal.org/project/search_api_federated_solr/issues/3191257

## Description

- Creates a 4.x release of Search API Federated Solr for use with Search API Solr 4.x and Solr 7.

## Testing instructions

- Checkout this branch
- If you have one, delete the `src` directory from the project root.
- Run `vagrant up` from the project directory -- this should re-install all project dependencies.
- `vagrant ssh`
- `phing install-all` and check errors.

## Remaining Tasks

We get the following error on `phing install-d8` which suggests a schema mismatch.

```
 [success] Successfully enabled: federated_default_content
 [error]  Drupal\search_api_solr\SearchApiSolrException while indexing item entity:node/2:en: Solr endpoint http://localhost:8983/ bad request (400). ERROR: [doc=9g44bj-federated_search-entity:node/2:en] multiple values encountered for non multiValued field ss_federated_terms: [Age>Adult, Traits>Loyal] in Drupal\search_api_solr\SolrConnector\SolrConnectorPluginBase->handleHttpException() (line 1016 of /var/www/federated-search-demo.local/web/d8/docroot/modules/contrib/search_api_solr/src/SolrConnector/SolrConnectorPluginBase.php). 
```

### CODE REVIEW

**Comment by agentrickard** [comment]
Changing the 'Federated terms' field to `Fulltext` removes the error but breaks its usage, since it is now passed as `tm_X3b_en_federated_terms`.

It looks like the schema handling has changed in 4.x of Search API solr. Fields that were marked as `sm_` (e.g. `sm_site_name`) are now `ss_`, which breaks the React application.

We'll need to sort through how the fields are handled and find a best solution for this problem.

We may have to provide a field mapping from Solr to React. I don't know.

**Comment by agentrickard** [comment]
I have a temporary (?) solution here -- https://www.drupal.org/project/search_api_federated_solr/issues/3192223#comment-13961896

**Comment by Kbentham** [comment]
The d7 sites look great and the d8 sites error as described.

**Comment by agentrickard** [comment]
@Kbentham So I wonder if we should commit this or leave it as a separate branch right now so we can compare against the current 3.x version.

As I type this, I think we may want to keep it separate for testing the `sm_` issue against 3.x

**Comment by agentrickard** [comment]
I think this issue may resolve our problem -- https://www.drupal.org/project/search_api_solr/issues/3153385 -- checking now.

**Comment by agentrickard** [comment]
Well, that didn't work. https://www.drupal.org/project/search_api_solr/issues/3192669

**Comment by agentrickard** [comment]
See also https://www.drupal.org/project/search_api_solr/issues/3192669

File: build.xml [hunk 1]
```diff
@@ -263,7 +263,7 @@
             </then>
             <else>
               <mkdir dir="src" mode="777" />
-              <exec command="git clone --branch=8.x-3.x https://github.com/palantirnet/search_api_federated_solr.git" dir="src" checkreturn="true" />
+              <exec command="git clone --branch=4.x https://github.com/palantirnet/search_api_federated_solr.git" dir="src" checkreturn="true" />
               <exec command="git clone --branch=8.x-3.x https://github.com/palantirnet/search_api_field_map.git" dir="src" checkreturn="true" />
```
Comment (byrond) [inline build.xml:267]: Do we need to update this to `4.x`? I see that's now a requirement in Composer. https://github.com/palantirnet/search_api_federated_solr/blob/57540f44e48e761dc5fec05aafc53ea37d7f22ad/composer.json#L17

**Pull request closed without merging**